distribution/pull_v2_windows.go

```diff
@@ -91,8 +90,8 @@ func filterManifests(manifests []manifestlist.ManifestDescriptor, p ocispec.Plat
 			continue
 		}
 		// TODO(thaJeztah): should we also take the user-provided platform into account (if any)?
-		if strings.EqualFold("windows", manifestDescriptor.Platform.OS) {
-			if err := checkImageCompatibility("windows", manifestDescriptor.Platform.OSVersion); err != nil {
+		if manifestDescriptor.Platform.OS == "windows" {
+			if err := checkImageCompatibility(manifestDescriptor.Platform.OS, manifestDescriptor.Platform.OSVersion); err != nil {
```

Comment:
Speaking of duplication, `checkImageCompatibility` is called _again_ on the result of this function in pull_v2.go. How feasible would it be to unify `filterManifests` between this file and pull_v2_unix.go, which offloads the work to a `platforms.Only()`-created `platforms.Matcher` instance. Maybe we need to work on `platforms.Matcher` first to do that, including teaching it about Hyper-V isolation vs Process isolation?

(Also, sorry about "single comment" spam. This wasn't _intended_ to be a review, I was actually looking for a different PR entirely.)